### PR TITLE
alwaysOpen prop

### PR DESCRIPTION
**Pull request by dfagan2**
Introduces an `alwaysOpen` property to force the `<Overlay>` to render immediately and always show. This helps developers simulate an on-page results list with a little CSS trickery.

### CODE REVIEW

File: src/Typeahead.react.js [hunk 1]
```diff
@@ -101,8 +102,8 @@ class Typeahead extends React.Component {
         <Overlay
           {...overlayProps}
           container={bodyContainer ? document.body : this}
-          referenceElement={this._inputContainer}
-          show={isMenuShown}>
+          referenceElement={this._inputContainer ? this._inputContainer : alwaysOpen}
```
Comment (dfagan2) [inline src/Typeahead.react.js]: This is needed because upon first render, the referenceElement prop comes through to the `<Overlay>` as `undefined` due to a race condition with the ref instantiation above. On any further interaction to or within the plugin, the Overlay would pop open and respect the new property. But I found that even though it is `undefined` on first render, it does have access to the ref and can render, so this is a little shim to get past that nuance, and achieves a proper initial render state.

**Comment by ericgio** [comment]
@dfagan2: Thanks for the PR. I think this works for the general case, but it's unclear to me how it holds up against edge cases or cases where conditions conflict (eg: no results + no empty label should hide the menu). Some tests for both the general case and various conditions would be great.

This approach also puts logic for whether the menu is shown in 2 different places. I'd rather see it go in the `isShown` util, which has all the logic for showing/hiding the menu depending on props/state.

**Comment by ericgio** [comment]
`open` and `defaultOpen` props were added in v3.3.0
